src/data/clean_dataset.py

```diff
@@ -147,36 +147,12 @@ def kmeans (df):
     df_k=df[['Neighborhood','SalePrice']]
 
     filterwarnings('ignore')
+   
+    lista = ['NAmes','NAme','CollgCr','OldTown','Edwards','Somerst','Gilbert','NridgHt','Sawyer','NWAmes','SawyerW','BrkSide','Crawfor','Mitchel','NoRidge','Timber','IDOTRR','ClearCR','StoneBr','Blmngtn','SWISU','MeadowV','BrDale','Veenker','NPKVill','Blueste']
 
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['NAmes'], "0")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['NAme'], "1")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['CollgCr'],"2")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['OldTown'],"3")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['Edwards'],"4")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['Somerst'],"5")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['Gilbert'],"6")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['NridgHt'],"7")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['Sawyer'],"8")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['NWAmes'],"9")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['SawyerW'],"10")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['BrkSide'],"11")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['Crawfor'],"12")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['Mitchel'],"13")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['NoRidge'],"14")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['Timber'],"15")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['IDOTRR'],"16")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['ClearCr'],"17")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['StoneBr'],"18")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['Blmngtn'],"19")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['SWISU'],"20")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['MeadowV'],"21")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['BrDale'],"22")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['Veenker'],"23")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['NPkVill'],"24")
-    df_k['Neighborhood'] = df_k['Neighborhood'].replace(['Blueste'],"25")
-
-    df_k['Neighborhood'] = df_k['Neighborhood'].astype(int)
-
+    for index, value in enumerate(lista):
+        df_k['Neighborhood'] = df_k['Neighborhood'].replace(lista, index)
```

Comment:
Did this work properly? Maybe I'm wrong, but I believe it should be `replace([value], index)`.